nvdaHelper/vbufBase/storage.cpp

```diff
@@ -132,10 +136,20 @@ inline void outputEscapedAttribute(wostringstream& out, const wstring& text) {
 			default:
 			out << *it;
 		}
+		if(maxLength>0) {
+			count++;
+			if(count==maxLength) {
+				break;

```

Comment:
Is it worth logging a debugWarning when we truncate? This shouldn't happen too often and it's possible this might be useful when trying to debug some future (as yet unknown) issue.

Reply:
Agreed.  outputEscapedAttribute now returns the number of input characters outputted to the stream. matchattributes checks this length against the original string length and logs a debug warning with the name of the attribute if the length is truncated.